bridge-history-api/internal/logic/l1_event_parser.go

```diff
@@ -270,12 +271,17 @@ func (e *L1EventParser) ParseL1BatchEventLogs(ctx context.Context, logs []types.
 				log.Error("Failed to unpack FinalizeBatch event", "err", err)
 				return nil, err
 			}
-			l1BatchEvents = append(l1BatchEvents, &orm.BatchEvent{
-				BatchStatus:   int(btypes.BatchStatusTypeFinalized),
-				BatchIndex:    event.BatchIndex.Uint64(),
-				BatchHash:     event.BatchHash.String(),
-				L1BlockNumber: vlog.BlockNumber,
-			})
+			var startBatchIndex uint64 = 0
+			if lastFinalizedBatchIndex, ok := f.getLastFinalizedBatchIndex(); ok {
+				startBatchIndex = lastFinalizedBatchIndex + 1
+			}
+			for index := startBatchIndex; index <= event.BatchIndex.Uint64(); index++ {
```

Comment:
What about changing the pattern of SQL update to <= batch index, and only keeping the last event (with max batch index)? This can reduce the number of SQL update ops.

Reply:
I'm a little confused of your comment here. 
The code snippet has nothing to do with any sql action. The `getLastFinalizedBatchIndex` method is already a local cache.
The only sql action is needed when initializing the `L1FetcherLogic` struct

Reply:
I mean we can update the SQL like:
```
		case btypes.BatchStatusTypeFinalized:
			// After darwin, FinalizeBatch event signals a range of batches are finalized,
			// thus losing the batch hash info. Meanwhile, only batch_index is enough to find the target batch when finalizing batches.
			db = db.Where("batch_index <= ?", l1BatchEvent.BatchIndex)
			db = db.Where("batch_status != ?", btypes.BatchStatusTypeFinalized)
			updateFields["batch_status"] = btypes.BatchStatusTypeFinalized
			if err := db.Updates(updateFields).Error; err != nil {
				return fmt.Errorf("failed to update batch event, error: %w", err)
			}
```

Reply:
literally one BatchEvent represents a single record in the database which relates to a single batch.
I think it's better to keep the BatchEvent array compared to change the BatchEvent meaning.

When it comes to the db operations, it can be collected the boundaries of batch indexes before doing the update. Besides, doing a range update with `<=` in a for loop is not efficient anyway.

Reply:
Doing a range update with <= in a for loop is not efficient anyway.
because after Darwin, there may be several batches per update, if we use `batch_index <= a && batch_status != finalized`, this can reduce the number of RPCs.
yeah. it's low frequency compared with cross-chain messages updates, but previously this was a very useful optimization (we merged the one-by-one read queries into one, reducing rpc latencies).

Reply:
_Doing a range update with <= in a for loop is not efficient anyway._
because after Darwin, there may be several batches per update, if we use `batch_index <= a && batch_status != finalized`, this can reduce the number of RPCs.
yeah. it's low frequency compared with cross-chain messages updates, but previously this was a very useful optimization (we merged the one-by-one read queries into one, reducing rpc latencies).

the new approach:
1. only store the FinalizeBatch event (one time), instead of splitting it into multiple events.
2. update db once.

Reply:
I have a new implementation, pls have a look